src/Security/Authentication/test/AuthenticationMiddlewareTests.cs

```diff
@@ -48,6 +48,58 @@ public async Task OnlyInvokesCanHandleRequestHandlers()
             Assert.Equal(607, (int)response.StatusCode);
         }
 
+        [Fact]
+        public async Task CanUseAuthenticationResultFeature()
+        {
+            var ticket = new AuthenticationTicket(new ClaimsPrincipal(), "Default");
+            var result = AuthenticateResult.Success(ticket);
+            SimpleHandler.Result = result;
+            var builder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    app.UseAuthentication();
+                    app.Use((context, next) =>
+                    {
+                        var auth = context.Features.Get<IAuthenticationResultFeature>();
+                        Assert.True(auth.Result.Ticket.Principal == ticket.Principal);
+                        context.Response.StatusCode = 200;
+                        return Task.CompletedTask;
+                    });
+                })
+                .ConfigureServices(services => services.AddAuthentication(o =>
+                {
+                    o.AddScheme("Default", s =>
+                    {
+                        s.HandlerType = typeof(SimpleHandler);
+                    });
+                    o.DefaultScheme = "Default";
+                }));
+            var server = new TestServer(builder);
+            var response = await server.CreateClient().GetAsync("http://example.com/");
+            response.EnsureSuccessStatusCode();
+        }
+
+        [Fact]
+        public async Task CanUseAuthenticationResultFeatureWithNoDefaultScheme()
+        {
+            var builder = new WebHostBuilder()
```

Comment:
Host Builder 😄

Reply:
TestServer needs a IWebHostBuilder :shrug:

Reply:
No it doesn't, it just works different. Check the test server tests for an example.